src/components/numberInput/index.tsx

```diff
@@ -0,0 +1,172 @@
+   */
+  marginRight?: number;
+};
+
+function NumberInput(props: NumberInputProps, ref: any) {
+  const {
+    onChangeNumber,
+    initialNumber,
+    fractionDigits = 2,
+    // @ts-expect-error
+    locale = 'en',
+    style,
+    containerStyle,
+    leadingText,
+    leadingTextStyle,
+    marginLeft,
+    trailingText,
+    trailingTextStyle,
+    marginRight,
+    onChangeText,
+    ...others
+  } = props;
+  const [options, setOptions] = useState<Options>(generateOptions(locale, fractionDigits));
+  const [data, setData] = useState<NumberInputResult>();
+
+  useDidUpdate(() => {
+    setOptions(generateOptions(locale, fractionDigits));
+  }, [locale, fractionDigits]);
+
+  const handleInitialValueChange = () => {
+    const newData = getInitialResult(options, initialNumber);
+    onChangeNumber(newData);
+    setData(newData);
+  };
+
+  useEffect(() => {
+    handleInitialValueChange();
+  }, []);
+
+  useDidUpdate(() => {
+    handleInitialValueChange();
+  }, [initialNumber]);
+
+  const hasText = useMemo(() => {
+    // Render both leading and trailing accessories so the text is centered AND the margin between the text and the accessories is correct
+    return !isEmpty(leadingText) || !isEmpty(trailingText);
+  }, [leadingText, trailingText]);
+
+  const leadingAccessoryStyle: TextProps['style'] = useMemo(() => {
+    return [{textAlign: 'right'}, leadingTextStyle, marginLeft ? {marginLeft} : undefined];
+  }, [leadingTextStyle, marginLeft]);
+
+  const leadingAccessory = useMemo(() => {
+    if (hasText) {
+      return <Text style={[styles.accessory, leadingAccessoryStyle]}>{leadingText}</Text>;
+    }
+  }, [hasText, leadingText, leadingAccessoryStyle]);
+
+  const trailingAccessoryStyle: TextProps['style'] = useMemo(() => {
+    return [trailingTextStyle, marginRight ? {marginRight} : undefined];
+  }, [trailingTextStyle, marginRight]);
+
+  const trailingAccessory = useMemo(() => {
+    if (hasText) {
+      return <Text style={[styles.accessory, trailingAccessoryStyle]}>{trailingText}</Text>;
+    }
+  }, [hasText, trailingText, trailingAccessoryStyle]);
+
+  const _style = useMemo(() => {
+    return [style, leadingAccessory ? undefined : {marginLeft}, trailingAccessory ? undefined : {marginRight}];
+  }, [style, leadingAccessory, marginLeft, trailingAccessory, marginRight]);
+
+  const _containerStyle = useMemo(() => {
+    return [styles.containerStyle, containerStyle];
+  }, [containerStyle]);
+
+  const _onChangeText = useCallback((text: string) => {
+    const newData = parseInput(text, options);
+    onChangeNumber(newData);
+    setData(newData);
+    onChangeText?.(text);
+  },
+  [onChangeNumber, options, onChangeText]);
+
+  const value = useMemo(() => {
+    return data?.type === 'valid' || data?.type === 'error' ? data.userInput : '';
+  }, [data]);
+
+  const formatter = useCallback(() => {
+    return data?.type === 'valid' ? data.formattedNumber : data?.type === 'error' ? data.userInput : '';
```

Comment:
That results with a TS error (because `data` is a Discriminated Union, which seems to not fully support this (yet?)).

Reply:
So you can add `userInput` to the empty state as well, WDYT?

Reply:
I don't really like it, it'll force empty to be inserted as:
```
export const EMPTY: NumberInputData = {type: 'empty', userInput: ''};
```
Instead of:
```
export const EMPTY: NumberInputData = {type: 'empty'};
```

BTW, here we still need the `data.formattedNumber` AFAIK